device-connectors/src/testflinger_device_connectors/devices/iotscript/iot-auto-sanity/sanity/agent/deploy.py

```diff
@@ -0,0 +1,539 @@
+            while True:
+                mesg = con.read_con()
+                if mesg.find("Fastboot:") != -1:
+                    con.write_con_no_wait()
+                    con.write_con_no_wait("fastboot usb 0")
+                    if syscmd("sudo uuu uc.lst") != 0:
+                        return {
+                            "code": FAILED,
+                            "mesg": f"{dev_data.project}"
+                            f" auto sanity was failed, deploy failed.",
+                        }
+
+                    con.write_con_no_wait("\x03")  # ctrl+c
+                    con.write_con_no_wait("run bootcmd")
+                    break
+
+        case "seed_override":
+            login(con)
+            ADDR = get_ip(con)
+            if ADDR == FAILED:
+                return {
+                    "code": FAILED,
+                    "mesg": f"{dev_data.project} auto sanity was failed,"
+                    f"target device DHCP failed.",
+                }
+
+            if check_net_connection(ADDR) == FAILED:
+                return {
+                    "code": FAILED,
+                    "mesg": f"{dev_data.project} auto sanity was failed,"
+                    f"target device connection timeout.",
+                }
+
+            if update_boot_assets:
+                boot_assets_update(ADDR)
+
+            scp_cmd = (
+                'scp -r -o "UserKnownHostsFile=/dev/null" '
+                '-o "StrictHostKeyChecking=no"'
+            )
+            if (
+                syscmd(
+                    "sshpass -p {} {} seed {}@{}:~/".format(
+                        dev_data.device_pwd,
+                        scp_cmd,
+                        dev_data.device_uname,
+                        ADDR,
+                    )
+                )
+                != 0
+            ):
+                print("Upload seed file failed")
+                return {"code": FAILED}
+
+            con.write_con("cd ~/")
+            con.write_con(
+                "cd /run/mnt/ubuntu-seed && sudo ls -lA | "
+                "awk -F':[0-9]* ' '/:/{print $2}' |"
+                "xargs -i sudo rm -fr {} && cd ~/"
+            )
+            con.write_con(
+                "cd seed/ && ls -lA | "
+                "awk -F':[0-9]* ' '/:/{print $2}' | "
+                "xargs -i sudo cp -fr {} /run/mnt/ubuntu-seed/ && cd ~/"
+            )
+            # We don't wait for prompt due to system could
+            # possible reboot immediately without prompt
+            con.write_con_no_wait(
+                "sudo snap reboot --install {}".format(
+                    os.path.relpath(
+                        str(glob.glob("seed/systems/[0-9]*")[0]),
+                        "seed/systems",
+                    )
+                )
+            )
+            con.write_con_no_wait("sudo reboot")
+
+        case "seed_override_lk":
+            login(con)
+            ADDR = get_ip(con)
+            if ADDR == FAILED:
```

Comment:
based on how it looks like get_ip() and check_net_connection() are getting used, I think you should consider just passing the exception and catching as specific of an exception as possible, then use logging to print the error and exit or raise as necessary so that calling code can do the right thing with it. I don't feel like replacing the exception handling with SUCCESS/FAILED is really making it much easier to read or follow.